src/Orleans.Core/Statistics/ClientStatisticsManager.cs

```diff
@@ -35,17 +34,15 @@ public ClientStatisticsManager(ClientConfiguration config, SerializationManager
             ApplicationRequestsStatisticsGroup.Init(config.ResponseTimeout);
         }
 
-        internal async Task Start(StatisticsProviderManager statsManager, IMessageCenter transport, GrainId clientId)
+        internal async Task Start(IMessageCenter transport, GrainId clientId)
         {
             runtimeStats.Start();
 
             // Configure Metrics
-            IProvider statsProvider = null;
             if (!string.IsNullOrEmpty(statisticsOptions.ProviderName))
             {
                 var extType = statisticsOptions.ProviderName;
-                statsProvider = statsManager.GetProvider(extType);
-                var metricsDataPublisher = statsProvider as IClientMetricsDataPublisher;
+                IClientMetricsDataPublisher metricsDataPublisher = this.serviceProvider.GetServiceByName<IClientMetricsDataPublisher>(extType);
```

Comment:
hmmm where is the code which register dataPublisher as `IClientMetricDataPublisher` named service into DI? I didn't see such code in `LegacyProviderConfigurator`. If it is not registered, then we cannot get service here right?

Reply:
I suspect you're correct.  Will look into.